File: Data Structures & Algorithms/longest-common-prefix/submission-2.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        int maxi=INT_MAX;
        string s1=strs[0];
        if(strs.size()==1) return strs[0];
        for(int i=1;i<strs.size();i++){
           
           string s2=strs[i];
           int cnt=0;
           string comm="";
           for(int j=0;j<min(s1.size(),s2.size());j++){
               
               if(s1[j]!=s2[j]){
                
                break;
               }
               else{
                 cnt++;
                 comm+=s1[j];

               }
           }
           if(maxi>cnt){
            ans=comm;
            maxi=cnt;
           }
 

        }


        return ans;

    }
};

File: Data Structures & Algorithms/time-needed-to-buy-tickets/submission-2.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
         int cnt=0;
        //  while(tickets[k]!=0){


        //     for(int i=0;i<tickets.length;i++){
                
        //         if(tickets[i]!=0){
        //             cnt++;
        //             tickets[i]--;
        //         }
        //         if(tickets[k]==0){
        //             return cnt;
        //         }


        //     }






        //  }

        for(int i=0;i<tickets.length;i++){

             if(i<=k){
                cnt+=Math.min(tickets[k],tickets[i]);
             }
             else{
                cnt+=Math.min(tickets[i],tickets[k]-1);
             }




        }




         return cnt;


    }
}